cache/lru/lru.go

```diff
@@ -155,24 +155,53 @@ func (c *Cache) Clear() {
 
 // Range call f on every key
 func (c *Cache) Range(f func(key Key, value interface{}) bool) {
-	c.mutex.Lock()
-	defer c.mutex.Unlock()
+	var (
+		preCache []entry
+	)
+	c.mutex.RLock()
 	for _, e := range c.cache {
-		kv := e.Value.(*entry)
-		if !f(kv.key, kv.value) {
+		if e == nil {
```

Comment:
this if{} is not necessary?

it is of course always safe to nil check, but the implementation guarantees that `e != nil`